Backpack-SwiftUI/Snippet/Classes/BPKSnippet.swift

```diff
@@ -0,0 +1,104 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright © 2022 Skyscanner Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKSnippet: View {
+    let image: Image
+    let accessibilityLabel: String
+    let headline: String?
+    let description: String?
+    let bodyText: String?
+    let imageOrientation: ImageOrientation
+    let onClick: (() -> Void)?
+    
+    public init(
+        image: Image,
+        accessibilityLabel: String,
+        headline: String? = nil,
+        description: String? = nil,
+        bodyText: String? = nil,
+        imageOrientation: ImageOrientation = .landscape,
+        onClick: (() -> Void)? = nil
+    ) {
+        self.image = image
+        self.accessibilityLabel = accessibilityLabel
+        self.headline = headline
+        self.description = description
+        self.bodyText = bodyText
+        self.imageOrientation = imageOrientation
+        self.onClick = onClick
+    }
+    
+    public var body: some View {
+        VStack(alignment: .leading, spacing: .md) {
+            RoundedRectangle(cornerRadius: .md, style: .continuous)
+                .foregroundColor(.clear)
+                .background(
+                    image
+                        .resizable()
+                        .background(.surfaceHighlightColor)
+                        .aspectRatio(contentMode: .fill)
+                        .accessibilityLabel(accessibilityLabel)
+                )
+                .aspectRatio(aspectRatio(for: imageOrientation), contentMode: .fit)
+                .clipShape(RoundedRectangle(cornerRadius: .md, style: .continuous))
+            if let headline {
+                BPKText(headline, style: .heading4)
+                    .lineLimit(2)
+                    .accessibilityAddTraits(.isHeader)
+            }
+            if let description {
+                BPKText(description, style: .subheading)
+                    .padding(.bottom, .xl)
+            }
+            if let bodyText {
+                BPKText(bodyText, style: .bodyDefault)
+                    .lineLimit(5)
+            }
+        }
+        .frame(maxWidth: .infinity)
+        .onTapGesture {
+            onClick?()
+        }
```

Comment:
Using `onTapGesture`, although something like a `Button` could provide more A11y functionality out of the box (such as voiceover indication that the card is tappable as a whole). Happy to re-vise.

Reply:
I agree that given that the whole snippet is tappable (and not just a section of it) the tap gesture handler could be set from outside directly. Would we need to coordinate this across platforms? 🤔

Reply:
🤔 Interesting. I would argue for keeping it, as it would make logic for tapping (which is more related to views, and thus to the domain of Backpack) more unified. Otherwise it could encourage a pattern where this component is used with `onTapGesture` in some places, or nested inside a `Button` in others, which knowing how SwiftUI behaves and changes across iOS versions, platforms and devices could save us some unexpected results.

Reply:
Update: it seems like the `onClick` function might have been more aimed for the desktop version, let's double check with design 👍

Reply:
Double checked, indeed it is specified to include the `onClick` action, even on apps